src/Aspire.Hosting.Docker/DockerComposeEnvironmentResource.cs

```diff
@@ -62,4 +62,9 @@ private Task PublishAsync(PublishingContext context)
 
         return dockerComposePublishingContext.WriteModelAsync(context.Model, this);
     }
+
+    internal void AddEnvironmentVariable(string name, string? description = null, string? defaultValue = null)
+    {
+        CapturedEnvironmentVariables[name] = (description, defaultValue);
```

Comment:
I don't think there's ever a case where we'll unintentionally override with duplicates. The only scenario I could think of would be if the parameter resource names were `parameter-1` and `parameter_1` and we don't permit underscores in resource names so I think this is fool-proof?

Reply:
There might be edge cases we need to deal with (a parameter named resource_image) or something crazy where the container image overrides the parameter env. I haven't thought through it much 😄